lib/imagemounter/utils/list_developer_image.js

```diff
@@ -0,0 +1,184 @@
+            return entry;
+        }
+        if (statResult.isDirectory()) {
+            const subFolderResult = await findDeveloperImageFromDirectory(fullPath);
+            if (subFolderResult) {
+                return subFolderResult;
+            }
+        }
+    }
+    return undefined;
+}
+
+/**
+ * @typedef {Object} ImagePath
+ * @property {string} developerImage
+ * @property {string} developerImageSignature
+ */
+
+/**
+ * Find developer image for certain version. If developer image does not exists,
+ * this will try to find and download developer image, unzip to `~/.appium/iOSSupport/`
+ * @param {string} version full version of iOS device. The first two parts of version
+ * will be preserved when sending the request, e.g. `14.7.1` will be changed to `14.7`
+ * @returns {ImagePath}
+ * @throws If developer image is not found, or error while downloading or unzipping.
+ */
+async function findDeveloperImage(version) {
+    const finalVersion = version.split('.').splice(0, 2).join('.');
+    const fileName = `${finalVersion}.zip`;
+    const DEFAULT_IMAGE_DIR = joinPath(await env.resolveAppiumHome(), DEFAULT_IMAGE_DIR_NAME);
+    const fullDownloadPath = joinPath(DEFAULT_IMAGE_DIR, fileName);
+    if (!await exists(DEFAULT_IMAGE_DIR)) {
+        await mkdir(DEFAULT_IMAGE_DIR, { recursive: true });
+    }
+    if (!await exists(fullDownloadPath)) {
+        await searchAndDownloadDeveloperImageFromGithub(finalVersion, fullDownloadPath);
+    }
+    const decompressPath = joinPath(DEFAULT_IMAGE_DIR, finalVersion);
+    let developerImageParentFolder = null;
+    if (await exists(decompressPath)) {
+        developerImageParentFolder = await findDeveloperImageFromDirectory(decompressPath);
+    }
+    if (!developerImageParentFolder) {
+        await zip.extractAllTo(fullDownloadPath, decompressPath);
+        developerImageParentFolder = await findDeveloperImageFromDirectory(decompressPath);
+    }
+    if (!developerImageParentFolder) {
+        throw new Error(`Unable to find unzipped developer image in ${decompressPath}`);
+    }
+    return {
+        developerImage: joinPath(developerImageParentFolder, DEVELOPER_IMAGE_FILE_NAME),
+        developerImageSignature: joinPath(developerImageParentFolder, DEVELOPER_IMAGE_SIGNATURE_FILE_NAME)
+    };
+}
+
+async function searchAndDownloadDeveloperImageFromGithub(finalVersion, fullDownloadPath) {
+    const fileNameRegExp = new RegExp(`${finalVersion}(\\(([\\w_|.()])+\\))?.zip`);
+    let fileUrl = null;
+    for (const repo of AVAILABLE_REPOS) {
+        const fileList = await listGithubImageList(repo.user, repo.name, repo.subFolderList);
+        if (!fileList) {
+            continue;
+        }
+        const targetFile = _.find(fileList, (item) => fileNameRegExp.test(item.path));
+        if (targetFile) {
+            fileUrl = `https://raw.githubusercontent.com/${repo.user}/${repo.name}/master/${repo.subFolderList.join('/')}/${targetFile.path}`;
```

Comment:
we should fetch a blob from `master`.  use a SHA1.  in lieu of maintaining our own repo of whatever this is, we can just update the SHA1.

Reply:
I don't quite sure how to do that. The original github api do contains a sha, should I check it after file downloaded?

Reply:
I've tried on this. There are no sha1 checksum on github. Maybe you mean we can just make a repo of sha1 of all version? This could be tough when maintaining beta version of iOS